types/wlr_surface.c

```diff
@@ -126,8 +126,12 @@ static void surface_set_opaque_region(struct wl_client *client,
 }
 
 static void surface_set_input_region(struct wl_client *client,
-		struct wl_resource *resource, struct wl_resource *region_resource) {
+		struct wl_resource *resource,
+		struct wl_resource *region_resource) {
 	struct wlr_surface *surface = wlr_surface_from_resource(resource);
+	if ((surface->pending.committed & WLR_SURFACE_STATE_INPUT_REGION)) {
```

Comment:
What is this doing?

Reply:
I am honestly not entirely sure myself, I just added it because the same thing is done for input regions a few lines above, and I just assumed that this not having it was a bug.
Perhaps though this is entirely unnecessary now that I think about it; both branches after that write to the region, making the clear unnecessary. It seemed to work fine before, so I'll remove it here and also for opacity regions.